ivy/src/main/scala/sbt/internal/librarymanagement/IvyActions.scala

```diff
@@ -142,37 +81,65 @@ object IvyActions {
         file
       )
       log.info("Wrote " + file.getAbsolutePath)
+      file
     }
   }
 
-  def deliver(module: IvySbt#Module, configuration: DeliverConfiguration, log: Logger): File = {
-    import configuration._
+  def deliver(module: IvySbt#Module, configuration: PublishConfiguration, log: Logger): File = {
+    val deliverIvyPattern = configuration.deliverIvyPattern
+      .getOrElse(sys.error("deliverIvyPattern must be specified."))
+    val status = getDeliverStatus(configuration.status)
     module.withModule(log) {
-      case (ivy, md, default) =>
+      case (ivy, md, _) =>
         val revID = md.getModuleRevisionId
         val options = DeliverOptions.newInstance(ivy.getSettings).setStatus(status)
-        options.setConfs(IvySbt.getConfigurations(md, configurations))
+        options.setConfs(getConfigurations(md, configuration.configurations))
         ivy.deliver(revID, revID.getRevision, deliverIvyPattern, options)
         deliveredFile(ivy, deliverIvyPattern, md)
     }
   }
+
+  def getConfigurations(
+      module: ModuleDescriptor,
+      configurations: Option[Vector[ConfigRef]]
+  ): Array[String] =
+    configurations match {
+      case Some(confs) => (confs map { _.name }).toArray
+      case None        => module.getPublicConfigurationsNames
+    }
+
   def deliveredFile(ivy: Ivy, pattern: String, md: ModuleDescriptor): File =
     ivy.getSettings.resolveFile(
       IvyPatternHelper.substitute(pattern, md.getResolvedModuleRevisionId)
     )
 
   def publish(module: IvySbt#Module, configuration: PublishConfiguration, log: Logger): Unit = {
-    import configuration._
+    val resolverName = configuration.resolverName match {
+      case Some(x) => x
+      case _       => sys.error("Resolver name is not specified")
+    }
+
+    // Todo. Fix publish ordering https://github.com/sbt/sbt/issues/2088#issuecomment-246208872
+    val ivyFile: Option[File] =
+      if (configuration.publishMavenStyle) None
```

Comment:
Isn't this a blocker for this PR? Does this (with the todo above) mean we no longer publish POM files?